runtime/src/lib.rs

```diff
@@ -87,16 +88,12 @@ pub trait Runner {
 pub trait Spawner: Clone + Send + Sync + 'static {
     /// Enqueues a task to be executed.
     ///
-    /// Label can be used to track how many instances of a specific type of
-    /// task have been spawned or are running concurrently (and is appended to all
-    /// metrics). Label is automatically appended to the parent task labels (i.e. spawning
-    /// "fun" from "have" will be labeled "have_fun").
-    ///
     /// Unlike a future, a spawned task will start executing immediately (even if the caller
     /// does not await the handle).
-    fn spawn<F, T>(&self, label: &str, f: F) -> Handle<T>
```

Comment:
It may make sense to revert the reference to self here (where spawn consumes the runtime).

I like it this way because it prevents someone from accidentally reusing a runtime).

Reply:
I don't understand this well enough to have an opinion TBH.   Is the idea a runtime should be used to spawn only one task?

Reply:
That's a good way to frame what this PR would now enforce...

Reply:
I thought about this a bit more and I like that spawning a task must consume some context.

I feel good saying "any spawned task must have some context" and "context cannot be shared between any two tasks" and "context doesn't come from nowhere".

Reply:
yeah this LGTM now that I understand, plus like the idea of a spawn_ref for those cases that don't need to clone.